### ISSUE TITLE
About SWA

### ISSUE THREAD

**Issue by yir4** [dnap512/SelfReg#2]
Hi, could you please provide the code of using SWA? Thanks.

**Comment by dnap512** [comment]
Hi, @yir4.

Thank you for your interest in our work.
This code includes SWA.
https://github.com/dnap512/SelfReg/blob/main/codes/train.ipynb

Thank you.

**Comment by yir4** [comment]
Hi, @dnap512 ,
Thanks for your reply.
1, I didn't find torch.optim.swa_utils.update_bn(loader, swa_model) to update BatchNorm running_mean. Did you do that on purpose?
2,  Do you use SWA from the begining or near the end of the training?

Thanks.

**Comment by dnap512** [comment]
We used this repository codes in our experiments.

1. Please ref this code.
https://github.com/dnap512/SelfReg/blob/05d9777139309f2495bd1e9394d73f4e56402744/codes/utils.py#L571

2. We use SWA per 1 epoch from 80% of training process to the end.

Thank you.
